hooks/pre-exit

```diff
@@ -160,16 +161,54 @@ find_and_upload() {
       exit "${BUILDKITE_PLUGIN_TEST_COLLECTOR_MISSING_ERROR:-1}"
     fi
   else
+    declare -a uploads_in_progress=()
+    echo "Uploading '${#matching_files[@]}' files matching '${FILES_PATTERN}'"
+    
     # needs to be part of else for bash4.3 compatibility
     for file in "${matching_files[@]}"; do
-      echo "Uploading '$file'..."
-      if ! upload "$TOKEN_VALUE" "$FORMAT" "${file}"; then
-        echo "Error uploading, will continue"
-      fi
-      if [[ "$ANNOTATE" != "false" ]]; then
-        save-report-url "${CURL_RESP_FILE}"
-      fi
+      iterations_waited=0
+      while [[ "${#uploads_in_progress[@]}" -ge $UPLOAD_CONCURRENCY ]]; do
+        iterations_waited=$((iterations_waited + 1))
+        if [[ "${DEBUG}" == "true" ]]; then
+          echo "Waiting for uploads to finish..."
+        fi
+
+        sleep 1
+
+        for index in "${!uploads_in_progress[@]}"; do
+          if ! ps -p "${uploads_in_progress[index]}" > /dev/null; then
+            unset 'uploads_in_progress[index]'
+          elif [[ "$iterations_waited" -gt 10 ]]; then
```

Comment:
I believe that this `10` should be `BUILDKITE_PLUGIN_TEST_COLLECTOR_TIMEOUT` instead... or be configurable through another plugin option

Reply:
Yep, that was definitely an oversight on my part. Using the standard timeout seems reasonable for the moment I think.